src/marshmallow/fields.py

```diff
@@ -1471,7 +1471,9 @@ def _serialize(self, value, attr, obj, **kwargs):
         if value is None:
             return None
         base_unit = dt.timedelta(**{self.precision: 1})
-        return int(value.total_seconds() / base_unit.total_seconds())
+        delta = round(decimal.Decimal(value.total_seconds()), 9)
+        unit = round(decimal.Decimal(base_unit.total_seconds()), 9)
```

Comment:
Should we add a comment to explain where the `9` comes from?

Also I'm wondering if it could be worth computing `unit` at init for each precision. Or even hardcode all of them. But this might not be worth the trouble.

Reply:
Reverting to integer arithmetic ended up being simpler than I anticipated. See https://github.com/marshmallow-code/marshmallow/issues/1865#issuecomment-904787313.